src/BrightScriptDebugSession.ts

```diff
@@ -287,6 +306,157 @@ export class BrightScriptDebugSession extends DebugSession {
         }
     }
 
+    private componentLibraryPostfix: string = '__lib';
+
+    protected async prepareAndHostComponentLibraries(componentLibraries, componentLibrariesOutDir: string, port: number) {
+        if (componentLibraries && componentLibrariesOutDir) {
+            this.componentLibrariesOutDir = componentLibrariesOutDir;
+            this.componentLibrariesStagingDirPaths = [];
+            let replace = require('replace-in-file');
+            let libraryNumber: number = 0;
+
+            // #region Prepare the component libraries and create some name spacing for debugging
+            for (const componentLibrary of componentLibraries as any) {
+                libraryNumber ++;
+                componentLibrary.outDir = componentLibrariesOutDir;
+                let stagingFolder = await this.rokuDeploy.prepublishToStaging(componentLibrary);
+                let paths = glob.sync(path.join(stagingFolder, '**/*'));
+                let pathDetails: object = {};
+
+                // Add breakpoint lines to the staging files and before publishing
+                this.sendDebugLogLine('Adding stop statements for active breakpoints in Component Libraries');
+                this.convertBreakpointPaths(componentLibrary.rootDir, componentLibrary.rootDir);
+                await this.addBreakpointStatements(stagingFolder, componentLibrary.rootDir);
+
+                for (let filePath of paths) {
+                    //make the path relative (+1 for removing the slash)
+                    let relativePath = filePath.substring(stagingFolder.length + 1);
+                    let parsedPath = path.parse(relativePath);
+
+                    if (parsedPath.ext) {
+                        let originalRelativePath = relativePath;
+
+                        if (parsedPath.ext === '.brs') {
+                            // Create the new file name to be used
+                            let newFileName: string = `${parsedPath.name}${this.componentLibraryPostfix}${libraryNumber}${parsedPath.ext}`;
+                            relativePath = path.join(parsedPath.dir, newFileName);
+
+                            // Update all the file name references in the library to the new file names
+                            replace.sync({
+                                files: [
+                                    path.join(stagingFolder, '**/*.xml'),
+                                    path.join(stagingFolder, '**/*.brs')
+                                ],
+                                from: (file) => new RegExp(parsedPath.base, 'gi'),
+                                to: newFileName
+                            });
+
+                            // Rename the brs files to include the postfix name spacing tag
+                            fs.rename(filePath, path.join(stagingFolder, relativePath), function(err) {
```

Comment:
These callbacks are not awaited, so technically we could encounter race conditions where the files haven't been totally renamed yet. 

In addition to the previous request to make this for loop async/parallel, you should consider using a promise-based rename function, and awaiting it. Check out the `move` function from the [fs-extra](https://github.com/jprichardson/node-fs-extra/blob/master/docs/move.md) package. You can do something like this:

```typescript
await fsExtra.move(source, dest);
```